devito/ir/support/basic.py

```diff
@@ -312,9 +333,9 @@ def distance(self, other, findex=None):
         # * If mismatching `directions`, set the distance to infinity
         # * If direction is Backward, flip the sign
         ret = []
-        for i, d0, d1 in zip(distance, self.directions, other.directions):
-            if d0 is d1:
-                ret.append(-i if d0 is Backward else i)
+        for i, it0, it1 in zip(distance, self.itintervals, other.itintervals):
+            if it0.direction is it1.direction and it0.interval == it1.interval:
```

Comment:
one 'is' and one '=='?

Reply:
yes. Directions are global objects (FORWARD, ANY, BACKWARD), while Intervals are objects, and we still don't have a caching mechanism for those, ie `x[0, 0]` and `x[0, 0]` may be two physically different objects, so `is` would fail (while `==` would clearly pass)